src/xenia/kernel/xam/xam_notify.cc

```diff
@@ -47,14 +55,15 @@ dword_result_t XNotifyGetNext(dword_t handle, dword_t match_id,
   }
 
   if (!id_ptr) {
-    return X_ERROR_INVALID_PARAMETER;
+    return X_E_INVALIDARG;
   }
   *id_ptr = 0;
+
   // Grab listener.
   auto listener =
       kernel_state()->object_table()->LookupObject<XNotifyListener>(handle);
   if (!listener) {
-    return X_ERROR_INVALID_HANDLE;
+    return 0;
```

Comment:
Are you sure about it? It seems kinda strange to return success when there is no listener

Reply:
Yes, in fact I forgot to correct an earlier return value.

Reply:
This isn't success, this is a 1/0 "got notification" and "didn't get notification".

Reply:
Ohh fact. I was looking (while making changes to it) at function inside which returns indeed error codes, but then converts it. My bad